src/back/PreviewServer.js

```diff
@@ -132,8 +132,8 @@ PreviewServer.prototype.hasProjectRoute = function (path) {
     return !!this._project_routes[path];
 };
 
-PreviewServer.prototype.serveProjectRoute = function (path, uri, res, project) {
-    return this._project_routes[path].call(this, uri, res, project);
+PreviewServer.prototype.serveProjectRoute = function (path, uri, req, res, project) {
+    return this._project_routes[path].call(this, uri, req, res, project, this);
```

Comment:
Ah yeah, not a very limpid one, good question :)
First is because we use it as context for `.call` (and one plugin may still override the context of the view func), and second is a parameter passed to the view func. Not very elegant, but we can see the first as pure bonus ("we need a context anyway, let's use that one").